### PR TITLE
1.19.3 support

### PR DESCRIPTION
**Pull request by datatags**
## Info
This PR adds support for 1.19.3

### Resolved changes
- Previously, NMS registries and registry keys were both stored in `Registry`. Now, the registries themselves are found in `BuiltInRegistries` while the keys are in `Registries`.
- `GoalSelector#removeAllGoals` now accepts a predicate for what goals to remove
- `Registry#ownedRegistryOrThrow` seems to have been replaced by `Registry#registryOrThrow` despite a different method with that name existing previously.

### Draft and unresolved changes
See review below

## Tested Environments

The project compiles, I've not done any testing yet because there are still changes that need to be made before it's ready.

## Demonstration
None yet...

### CODE REVIEW

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java [hunk 1]
```diff
@@ -0,0 +1,1565 @@
...
+    }
+
+    @Override
+    public AttributeInstance getAttributeInstance(Mob m, Attribute a) {
+        net.minecraft.world.entity.ai.attributes.Attribute nmsAttribute = BuiltInRegistries.ATTRIBUTE.get(toNMS(a.getKey()));
+        return new AttributeInstance1_19_R2(a, toNMS(m).getAttribute(nmsAttribute));
+    }
+
+    public static net.minecraft.world.entity.ai.gossip.GossipType toNMS(GossipType t) {
+        return net.minecraft.world.entity.ai.gossip.GossipType.valueOf(t.getKey().getKey().toUpperCase());
```
Comment (datatags) [inline 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:1399]: `GossipType.byId(String)` is gone, but the class now extends [Codec](https://kvverti.github.io/Documented-DataFixerUpper/snapshot/com/mojang/serialization/Codec.html). I assume Codec has the same functionality that `byId` did, but I can't figure out how to use it, so I've replaced it with `valueOf`.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java [hunk 2]
```diff
@@ -0,0 +1,1565 @@
...
+
+        return nmsValue;
+    }
+
+    public static Object fromNMS(Mob m, String key, Object nmsValue) {
+        Object value = nmsValue;
+
+        if (nmsValue instanceof GlobalPos l) {
+            BlockPos pos = l.pos();
+            World w = ((CraftServer) Bukkit.getServer()).getHandle().getServer().registries().compositeAccess().registryOrThrow(Registries.DIMENSION).get(l.dimension()).getWorld();
```
Comment (datatags) [inline 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:799]: The field `MinecraftServer#registryHolder` was replaced by `MinecraftServer#registries()`, but `registries()` returns a `LayeredRegistryAccess`. I've replaced it with `registries().compositeAccess()` but I don't know if that does the same thing or not.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java (hunk 1 above)
Comment (GamerCoder215) [reply 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:1399]: I didn't use `valueOf` because the enum name will obfuscate on runtime (because of the remapping plugin). I'll take a look at the source code when I get the chance.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java [hunk 3]
```diff
@@ -0,0 +1,1565 @@
...
+            case CRAMMING -> DamageSource.CRAMMING;
+            case DROWNING -> DamageSource.DROWN;
+            case STARVATION -> DamageSource.STARVE;
+            case CONTACT -> DamageSource.CACTUS;
+            case MAGIC -> DamageSource.MAGIC;
+            case FALL -> DamageSource.FALL;
+            case FLY_INTO_WALL -> DamageSource.FLY_INTO_WALL;
+            case VOID -> DamageSource.OUT_OF_WORLD;
+            case WITHER -> DamageSource.WITHER;
+            //case FALLING_BLOCK -> DamageSource.FALLING_BLOCK;
```
[the remark this answers is not in the document]
Comment (GamerCoder215) [reply 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:720]: Looking at the source code, it seems like that DamageSource requires an entity with the source.
I might change the parser to include the actual event in to input it, remove it entirely, or test it will a null value.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java (hunk 2 above)
Comment (GamerCoder215) [reply 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:799]: It seems to return a type of "Frozen" registry, but that might only be affected when it tries writing things. Seems to work.
I'll test this when I get home.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java [hunk 4]
```diff
@@ -1396,7 +1402,11 @@ public AttributeInstance getAttributeInstance(Mob m, Attribute a) {
     }
 
     public static net.minecraft.world.entity.ai.gossip.GossipType toNMS(GossipType t) {
```
Comment (GamerCoder215) [inline 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:1404]: Switched to a loop w/comparing the internal IDs.

File: 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java (hunk 2 above)
Comment (GamerCoder215) [reply 1_19_R2/src/main/java/me/gamercoder215/mobchip/abstraction/ChipUtil1_19_R2.java:805]: Seems like my original thought was correct: writing isn't allowed and requires hacky things to happen, but reading is perfectly ok, which is what we're trying to do.

**Comment by GamerCoder215** [comment]
Tested, and works pretty well competed to other versions.

I will note that I might add a method that sends update packets for body methods in the future, since they're happening server-side and not visually.
